mkosi

```diff
@@ -1023,10 +1024,24 @@ SigLevel    = Required DatabaseOptional
 """.format(args=args, server=server))
 
     subprocess.run(["pacman", "--color", "never", "--config", os.path.join(workspace, "pacman.conf"), "-Sy"], check=True)
+    # determine base packages list from base metapackage
     c = subprocess.run(["pacman", "--color", "never", "--config", os.path.join(workspace, "pacman.conf"), "-Sg", "base"], stdout=subprocess.PIPE, universal_newlines=True, check=True)
     packages = set(c.stdout.split())
     packages.remove("base")
 
+    other_official_kernel_packages = [
+        "linux-lts",
+        "linux-hardened",
+        "linux-zen"
+    ]
+
+    kernel_package = "linux"
+    for p in other_official_kernel_packages:
+        if p in packages:
+            kernel_package = p
+    if kernel_package != "linux":
+        packages.remove("linux")
```

Comment:
hmm, i figure python gurus could write this in a prettier way using the set logic they have, i.e. by determining the intersection of both lists or so... but given that I am not guru enough for knowing how to write that off the top of my head I'll take this as it is.

Reply:
I reworked the logic to support multiple kernels (and not just pick the first one), to keep "linux" if the user specified it explicitly, and use sets/intersection instead of the basic loop i used.